webapp/javascript/components/Continuous.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect } from 'react';
+import { useAppSelector, useAppDispatch } from '@webapp/redux/hooks';
+import {
+  reloadAppNames,
+  selectAppNames,
+  setQuery,
+} from '@webapp/redux/reducers/continuous';
+
+export default function Continuous({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const dispatch = useAppDispatch();
+  const appNames = useAppSelector(selectAppNames);
+
+  useEffect(() => {
+    async function loadAppNames() {
+      if (appNames.length <= 0) {
+        // Load application names
+        const names = await dispatch(reloadAppNames());
+
+        // Pick the first one
+        if (names.payload.length > 0) {
```

Comment:
This should only happen if query is empty.

We should also add `{}` to the end of it so that it's clear that you can add tags.

Reply:
Good catch, thanks!

RE adding `{}` this is tricky (when to add `{}`? when it's baked already?), so I went back and refactored the code to use nominal typing. Now the action expects a `Query` which you can then convert via `brandQuery(string) => String` or `queryFromAppName(string) => Query`